### PR TITLE
Task/refactor deposit manager

### PR DESCRIPTION
**Pull request by mattrussell36**
# Related issues 🔗

N/A

# Description ℹ️

- Adds a zustand store for deposit state
- Dont rely on useEffects to trigger refetch of balances
- Removes setting default asset via url as its not ever used
- Changes deposit related hooks to just return functions to fetch balances, rather than fetching everything on mount in multiple different hooks
- Uses a react-hook-form Controller component for the select asset dropdown

### CODE REVIEW

File: libs/deposits/src/lib/deposit-form.tsx [hunk 1]
```diff
@@ -154,16 +147,30 @@ export const DepositForm = ({
...
+              id="asset"
+              {...field}
+              onChange={(e) => {
+                field.onChange(e);
+                onSelectAsset(e.target.value);
+              }}
+            >
+              <option value="">{t('Please select')}</option>
+              {assets
+                .filter((a) => a.source.__typename === 'ERC20')
```
Comment (bglownia) [inline libs/deposits/src/lib/deposit-form.tsx:165]: I don't like this 'ERC20' literal. It's repeated couple times across project
I think we should have const and use it instead
```
export const ERC20 = 'ERC20';
export interface ERC20AssetSource {
  __typename: typeof ERC20;
  contractAddress: string;
}
if (asset?.source.__typename === ERC20) ...
```

Comment (mattrussell36) [reply libs/deposits/src/lib/deposit-form.tsx]: Moved types and added a isAssetTypeERC20 function to libs/react-helpers/assets. Saves some duplication too

File: libs/deposits/src/lib/deposit-form.tsx [hunk 2]
```diff
@@ -196,9 +203,9 @@ export const DepositForm = ({
           </UseButton>
         )}
       </FormGroup>
-      {selectedAsset && limits && (
+      {selectedAsset && max && deposited && (
```
[the remark this answers is not in the document]
Comment (mattrussell36) [reply libs/deposits/src/lib/deposit-form.tsx:205]: `max` and `deposited` in this case are `BigNumber | undefined`. So we don't need to worry about the falsy-ness of 0

**Review by MadalinaRaicu — APPROVED** [review]

File: libs/deposits/src/lib/deposit-form.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (bglownia) [reply libs/deposits/src/lib/deposit-form.tsx:205]: I'm not so sure about it
image

**Review by bglownia — APPROVED** [review]

**Pull request merged**